sdk/cosmos/azure-cosmos/azure/cosmos/auth.py

```diff
@@ -138,9 +144,166 @@ def __GetAuthorizationTokenUsingResourceTokens(resource_tokens, path, resource_i
 
         # Get the last resource id or resource name from the path and get it's token from resource_tokens
         for i in range(len(path_parts), 1, -1):
-            segment = path_parts[i-1]
+            segment = path_parts[i - 1]
             sub_path = "/".join(path_parts[:i])
             if not segment in resource_types and sub_path in resource_tokens:
                 return resource_tokens[sub_path]
 
     return None
+
+
+# pylint:disable=too-few-public-methods
+class _CosmosBearerTokenCredentialPolicyBase(object):
+    """Base class for a Cosmos Bearer Token Credential Policy.
+    Based on ~azure.core.pipeline.policies.BearerTokenCredentialPolicy.
```

Comment:
How does this implementation differ from the implementation in Core?

Reply:
Only in the authorization header, pushing changes now to reflect that.